src/controllers/objecting.entity.name.controller.ts

```diff
@@ -16,6 +22,65 @@ const validators = [
   check(DIVULGE_INFO_FIELD).not().isEmpty().withMessage(ErrorMessages.SELECT_TO_DIVULGE),
 ];
 
+const showPageWithSessionDataIfPresent = (session: Session, res: Response) => {
+  let existingName;
+  let yesChecked: boolean = false;
+  let noChecked: boolean = false;
+  const objectionCreate: ObjectionCreate = retrieveFromObjectionSession(session, SESSION_OBJECTION_CREATE);
+  if (objectionCreate) {
+    existingName = objectionCreate.fullName;
+    const existingSharedIdentity = objectionCreate.shareIdentity;
+    yesChecked = existingSharedIdentity;
+    noChecked = !existingSharedIdentity;
+  }
+
+  return res.render(Templates.OBJECTING_ENTITY_NAME, {
+    fullNameValue: existingName,
+    isYesChecked: yesChecked,
+    isNoChecked: noChecked,
+    templateName: Templates.OBJECTING_ENTITY_NAME,
+  });
+}
+
+const showPageWithMongoData = async (session: Session, res: Response, next: NextFunction) => {
+  try {
+    const objection: Objection = await getObjection(session);
+    const existingName: string = objection.created_by.fullName;
+    const existingShareIdentity: boolean = objection.created_by.shareIdentity;
+    if (existingName && existingShareIdentity !== undefined) {
+      return res.render(Templates.OBJECTING_ENTITY_NAME, {
+        fullNameValue: existingName,
+        isYesChecked: existingShareIdentity,
+        isNoChecked: !existingShareIdentity,
+        templateName: Templates.OBJECTING_ENTITY_NAME,
+      });
+    } else {
+      return next(new Error("Existing data not present"));
+    }
+  } catch (e) {
+    logger.error(e.message);
+    return next(e);
+  }
+}
+
+/**
+ * GET checks for change flag and renders page
+ * @param req
+ * @param res
+ * @param next
+ */
+export const get = async (req: Request, res: Response, next: NextFunction) => {
+  const session: Session | undefined = req.session as Session;
+  if (session) {
+    if (retrieveFromObjectionSession(session, CHANGE_ANSWER_KEY)) {
+      return await showPageWithMongoData(session, res, next);
+    } else {
+      return showPageWithSessionDataIfPresent(session, res);
+    }
+    return next(new Error("No Session present"));
```

Comment:
just spotted a bug i think, line 80 the return error for no session is inside the if(session) block meaning the session is present, and it would never be called anyway due to the if/else above returning the pages. I think line 80 should go between 81 and 82.